### DISCUSSION TITLE
sahi/slicing/slice_image/_export_single_slice Cannot use 'keep'. jpg not recognized as jpg. Related to commit ad25062dd0ee495796669b684754231416568933

### DISCUSSION THREAD

**Discussion by Greg7000** [obss/sahi#997] [Q&A] [answered]
Basically getting an error similar to https://github.com/python-pillow/Pillow/issues/4036 (
Cannot use 'keep'. jpg not recognized as jpg) even though my images folder points on a folder of .jpg.

Currently my fix is to replace:
    _def _export_single_slice(image: np.ndarray, output_dir: str, slice_file_name: str):
        image_pil = read_image_as_pil(image)
        slice_file_path = str(Path(output_dir) / slice_file_name)
        # export sliced image
        image_pil.save(slice_file_path, quality="keep")
        image_pil.close()  # to fix https://github.com/obss/sahi/issues/565
        verboselog("sliced image path: " + slice_file_path)_

With:
    _def _export_single_slice(image: np.ndarray, output_dir: str, slice_file_name: str):
        image_pil = read_image_as_pil(image)
        slice_file_path = str(Path(output_dir) / slice_file_name)
        # export sliced image
        **#image_pil.save(slice_file_path, quality="keep")**
        **image_pil.save(slice_file_path) # WORKAROUND**
        image_pil.close()  # to fix https://github.com/obss/sahi/issues/565
        verboselog("sliced image path: " + slice_file_path)_

which is not the best fix.... Can you assign a PR to fix this?

I am using 0.11.15 with pillow=10.0.0

**Comment by Greg7000** [comment]
This issue seem to be a consequence of the following commit: ad25062dd0ee495796669b684754231416568933 (5 nov 2023 5h45)

**Comment by fcakyon** [comment, marked as answer]
Thanks for bringing this issue up @Greg7000 !

Its now fixed with: https://github.com/obss/sahi/pull/1037
